### ISSUE TITLE
Rebuild or Refactor

### ISSUE THREAD

**Issue by kellyjandrews** [salesforce-marketingcloud/FuelSDK-Ruby#29]
Would love to get everyones opinion - should we start from the beginning, or refactor what we have? Seems like a lot of issue keep popping up, and this is compatible with Ruby 1.9.3 - might be nice to support the latest Ruby version. 

Any suggestions?

**Comment by kkohrt** [comment]
@kellyjandrews thanks for reaching out. I'm not sure a complete rewrite is needed. I'd be concerned about breaking existing code that is using the FuelSDK if  that happened. Instead, Some things I would put on my wish list are:

(1) Make the server side code more robust--we hit access limits all the time keeping our user data up to date in real-time, and had to resort to updating many things using the asynchronous FTP batch update on like a 4 hour schedule, which really takes deterministic processing out of the developers hands: there is no way to programmatically catch, report, and possibly remediate error conditions in the data. Badgeville, by way of comparison, almost never hits a throughput limit for us; and only ever complains because we have two event processors sending it data in parallel.
(2) more detailed error responses from the API
(3) w.r.t FuelSDK, maybe split the data synchronization and the email send functionality into two gems (though one may depend upon the other).

**Comment by michaelstalker** [comment]
Take a look at what @barberj has done with [his fork](https://github.com/barberj/fuelsdk) of the repo. It may be worth trying to merge his work with this repo. I think the design of the two repos has diverged, but it may be a good step in updating the current repository.

It would be nice to have only one repository that is both canonical and supported.
